### ISSUE TITLE
Web UI:  Change from TILES to LIST with checkboxes

### ISSUE THREAD

**Issue by davepl** [PlummersSoftwareLLC/NightDriverStrip#301]
Rather than small tiles for each effect, we should use a "list" format where each effect gets its own row tile. There should be a checkbox on the left side to enable/disable the effect.

Other functionality, such as Activate, can be a link within the tile, perhaps right-aligned.

![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/36e15367-c15b-4beb-a95c-98e7a2415bcd)

**Comment by Louis-Riel** [comment]
I wanted to start a convo regarding this. I already have 2 display modes, I will add this layout, and make it the default.

For the effect decorations, I was thinking that it would be nice to have some attributes added to the effect list to be used for decorations. 

I think maybe a picture, and now, looking at the sketch you put up there, a color marquee configuration.

**Comment by Louis-Riel** [comment]
This is the initial layout of this model, will refine over time:
Desktop
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/53452394/77a468bb-a6f7-4989-bdf3-f3bfdf1d944a)

Mobile
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/53452394/11be24e8-a80a-497a-8a76-c2aec49b62b7)

**Comment by Louis-Riel** [comment]
The light bar could have a configurable set of colors defined at the effect level if we add that. Also, the light bar will later be animated for show, I had issues making that work.

**Comment by davepl** [comment]
Cool!  Could the light “bar” also be a grid?  If so, the web page could pull the current contents of the strip or matrix from the ColorData server and display the current contents there….

**Comment by Louis-Riel** [comment]
I'm pretty sure the web interface cannot do raw sockets, but here's half a thought. The socket is meant to be used by some other thing running on the network. That thing could be running any framework server thing, and that could also expose a web socket. So the server could send color streams to the esp32, and report back to websocket. So the next thing is that the chip served site could use the UI build sequences, to then send that over the server,over the websocket, for it then play it back on the esp32. The server would then report the new running stream over websocket clients, so there, yes, the web UI could show a representative glow of the light bar.

**Comment by KeiranHines** [comment]
I did a quick and simple design change to implement this. In the top right of the 'designer' section now there is a toggle to switch between the old card layout and the new list layout. Clicking anywhere on the list item with popup the settings. 

Any feedback on the design would be great. Other its ready for a PR. 
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/befa8593-d488-4cba-9e5f-0c8f66953034)
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/e1bc4755-bf27-40f4-95d7-d102b2a52ddc)

**Comment by rbergen** [comment]
@KeiranHines I'm kind of speaking on Dave's behalf - which means I may be wrong - but I think he's looking for a list design that matches the ones shown in earlier comments in this issue (and the issue itself) a little more closely. 

My personal comments are:

- The current list item design looks a bit "cramped" to me; I think it would be easier to use this design if each of the list items has more height.
- I think we should include the settings cog in each of the list items, as we now do on the cards. I don't think opening the settings dialog by clicking anywhere on a list item is very intuitive, and I also don't think it'll be sustainable once the drag-and-drop reordering of effects (as per #300) is implemented.

I'll check with @davepl and ask him to comment directly with his concrete feedback on this iteration of the design.

**Comment by KeiranHines** [comment]
Thanks for the feedback. I went a bit back and forth between giving the rows more height and being a more compact layout. I can try a few different heights and paddings post screenshot. 

As for settings. I actually have click and drag to reorder working on this branch. It picks up the drag event separately from a single click for settings. I didn't add the cog initially cause it wasn't in the above designs and I wasn't really sure where I wanted to put it in the row. Again I will add it in and post some more screenshots.

**Comment by rbergen** [comment]
Thanks! I think the cogwheel may well not have been in the designs because the whole effect setting dialog hadn't been implemented yet - it certainly wasn't included in this repo's main branch.

**Comment by davepl** [comment]
Just thinking out loud, I'd imagine each element could render as a tile rather than as a simple horizontal entry, something like this from another "dark model control panel UI phone" that I googled for:

<img width="254" alt="image" src="https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/9d5f4aa6-95f3-4c0f-a448-7b2ff3ee9550">

But then the question is "what does it include?".  Right now we only have two attributes in the tile (name and enabled).  

```
----
NAME                 [Play Button] 
Description                    Time Remaining
----
```
Of course inactive ones wouldn't have Time Remaining, I suppose.

**Comment by rbergen** [comment]
I'm just quickly noting that any description beyond the title (or any other informative property or decoration) would constitute a `LEDStripEffect` property we'd have to add and expose to the web UI via the API. I guess it makes sense to turn it into an effect setting as well, so it can be changed via the UI if a user wants to do so.

**Comment by KeiranHines** [comment]
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/c410b93f-45a7-4d70-b0c7-2a20e07458f7)
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/d59a69d1-cef1-4640-93c4-0585507742d7)
A couple of small updates. I have increased the size a bit. I have not yet added the settings cog. I was thinking next to the effect name.

Thoughts?

**Comment by KeiranHines** [comment]
And with settings cog: 
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/c7c9191b-1168-4008-9a18-461c5dd283b6)
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/721f30cb-31ee-4c59-8e5d-87f942c5a966)

**Comment by davepl** [comment]
Cool!  At what point do you want me to start bugging you about layout?  Like the UI should work on the phone and desktop, and likely the tiles need a max width, that kind of thing.

I haven’t seen anything that can’t be changed later so far as I know, so I’ve been straying out of it!

Cheers,
Dave

PS:  Here’s a good example of the Nanoleaf UI with tiles:

**Comment by KeiranHines** [comment]
I have just pushed the latest updates to [this branch](https://github.com/KeiranHines/NightDriverStrip/tree/feature/designerImprovements) It is ready for any all feedback. It should be pretty mobile friendly. I haven't added a max width as the middle section of the row (name section) is set to grow to the available space meaning the rows should always been full width and the name should always be centered. On a large 4k screen or ultrawide this may mean the rows are a bit wide, but on such screen I prefer the grid layout anyway, hence why I left the option for both.

**Comment by KeiranHines** [comment]
@robertlipe #491 has been opened. I will address that after I finish up with the drag and drop and list view. Shouldn't be too hard to add.

**Comment by KeiranHines** [comment]
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/b9a52619-9883-4002-81fe-9bfe9a234bea)
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/11905a35-0b5b-462d-ad29-c7b7e6b5b99a)
Another idea with a bit of color and details added. I added the time remaining as either XX.X (h|m|s) depending on most significant time remaining and added a yellow color when less than half the time left, red when less than a quarter.

**Comment by KeiranHines** [comment]
@robertlipe  the time here is the remaining effect time. The same time remaining as in the top left, just formatted to be readable instead of just ms. 
I agree the yellow/red are more call to action colors. I couldn't really think of colors I liked more. I just felt it could use a splash of color.
As for linear progress vs circular. The current grid layout had linear. And I considered that. But as you noted. On a large screen, or a short timer it was comically large and raced across my screen so I switched it out for the circular.

**Comment by rbergen** [comment]
I'll mainly comment on the color of the countdown timer visualization, to the extent that we really shouldn't use signal colors for that. The neutral ➡️ yellow ➡️ red progression will make me instinctively panic about "how do I stop this?!?", which we all agree is not the point or intention. 

Beyond that, I don't have a very strong personal opinion about linear vs circular progress (both can work), but I do note we have a lot more horizontal space in the list layout than we have vertical, which may plea for a linear progress indicator after all. And yes, at the same time it should not become so big as to cause eyeball strain while tracking it. Maybe it's possible to put a few designs for horizontal indicators together, so we can make up our minds by looking at them?

**Comment by davepl** [comment]
I’ve been unable to do a filesystem upload… are there any special tricks, like holding down the BOOT button, or anything required these days?

- Dave

Serial port /dev/cu.usbserial-8411301
Connecting......

A fatal error occurred: Invalid head of packet (0x01): Possible serial noise or corruption.
*** [uploadfs] Error 2
================================================================================================================================================== [FAILED] Took 9.44 seconds ==================================================================================================================================================

Environment    Status    Duration
-------------  --------  ------------
mesmerizer     FAILED    00:00:09.442

(Pretty sure it’s no serial noise, I can flash fine…)

**Comment by KeiranHines** [comment]
I haven't had any issues building/uploading. I will try tonight and report back if I see similar.

**Comment by davepl** [comment]
Figured it out… serial monitor was open in the background, which prevents it from working!

I did a sync from the main tree and built and uploaded flash and filesystem, but don’t get the tiled version yet.  Or did I jump the gun, and this isn’t in main yet?

Thanks
Dave

**Comment by KeiranHines** [comment]
Not on main yet. It's on https://github.com/KeiranHines/NightDriverStrip/tree/feature/designerImprovements

**Comment by KeiranHines** [comment]
Examples with a linearProgress (Circular still there for comparison). Each one has a slightly different vertical position. 
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/e8667e48-58dd-45f6-9f86-756bd28e872f)
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/8f945aee-a32a-4e7c-966d-31dd47bc096d)
![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/2768974/3e5489a7-701c-4165-beba-67ba3d3d8d66)

**Comment by rbergen** [comment]
@KeiranHines Thanks! My vote goes to #3 as well. And now that I've seen it, I think I prefer it over the circular progress indicator after all - the latter now comes across to me as if it's pushed into a corner.

**Comment by KeiranHines** [comment]
I have updated my branch for this work to use option 3 over the circular progress. Once my current PR is merged (#494) I will open a PR for this unless there is any other feedback before opening it.
